dockerfile_parse/util.py

```diff
@@ -225,9 +225,15 @@ def extract_labels_or_envs(env_replace, envs, instruction_value):
         key_val_list.append((key, val))
 
     else:  # LABEL/ENV "name"="value"
-
+        last_item = None
```

Comment:
Should this be an empty str?

Reply:
no, this is just to initialize the variable: the first condition in the loop will always take the else branch

Reply:
..because a valid LABEL or ENV will always take a value, so shlex_splits_raw will always have at least one item, is that right?

Reply:
That was my reasoning, but on the other hand, someone can easily parse a broken dockerfile: in such case, do we just raise a ValueError?

Reply:
I think ideally we'd raise ValueError with a descriptive message, yes.

Reply:
I changed the initial value to "" as suggested by Luiz.

Unfortunately, `shlex.split` is too good. In order to make these crazy inputs work the same way as dockerd does, we would need change a lot of code. I'm not gonna do that in this PR. I don't think it's such a bad thing that we can parse more than dockerd. Worst case the build fails, no big deal.